gnomad_constraint/pipeline/constraint_pipeline.py

```diff
@@ -0,0 +1,135 @@
+# noqa: D100
+# cSpell: disable
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.utils.filtering import (
+    filter_x_nonpar,
+    filter_y_nonpar,
+)
+from gnomad.utils.reference_genome import get_reference_genome
+from gnomad_constraint.resources.resource_utils import (
+    get_sites_resource,
+    get_preprocessed_ht,
+    get_logging_path,
+    annotated_context_ht,
+    CURRENT_VERSION,
+    VERSIONS,
+    DATA_TYPES,
+)
+from gnomad_constraint.utils.constraint import (
+    add_vep_context_annotations,
+    prepare_ht_for_constraint_calculations,
+)
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Execute the constraint pipeline."""
+    hl.init(log="/constraint_pipeline.log")
+
+    test = args.test
+    overwrite = args.overwrite
+    # TODO: gnomAD v4 is still in production, for now this will only use 2.1.1.
+    version = args.version
+    if version not in VERSIONS:
+        version = CURRENT_VERSION
+        logger.warning(
+            "The requested version of resource Tables are not exist, will use v2 as default."
+        )
+
+    try:
+        if args.preprocess_data:
+            logger.info("Adding VEP context annotations...")
+            # TODO: Need to add function that annotates methylation, coverage, and gerp in the vep context table.
+            context_ht = annotated_context_ht.versions[version].ht()
+            # Add annotations used in constraint calculations.
+            for data_type in DATA_TYPES:
+                if data_type != "context":
+                    ht = get_sites_resource(data_type, version).ht()
+                else:
+                    ht = context_ht
+
+                # Filtering the Table to chr20, chrX, and chrY for testing if applicable
+                if test:
+                    rg = get_reference_genome(context_ht.locus)
+                    contigs_keep = [
+                        hl.parse_locus_interval(c, reference_genome=rg)
+                        for c in [rg.contigs[19], rg.x_contigs[0], rg.y_contigs[0]]
+                    ]
+                    logger.info(
+                        "Filtering the %s HT to chr20, chrX, and chrY for testing...",
+                        data_type,
+                    )
+                    ht = hl.filter_intervals(ht, contigs_keep)
+
+                # Add annotations from VEP context Table to genome and exome Tables.
+                preprocessed_ht = (
```

Comment:
since preprocessed_ht  is immediately reassigned to full_ht below and not used again, i think it might be more clear to just keep this named ht so that there are less table names to keep track of